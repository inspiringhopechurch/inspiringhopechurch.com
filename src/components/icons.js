import { library } from "@fortawesome/fontawesome-svg-core";
import {
  faAngleDown,
  faAngleUp,
  faBadgePercent,
  faCommentAltLines,
  faEnvelope,
  faHeart,
  faMapMarkerAlt,
  faUser,
  faPaperPlane,
  faPhone,
} from "@fortawesome/pro-solid-svg-icons";
import {
  faFacebook,
  faFacebookF,
  faTwitter,
  faLinkedinIn,
  faReact,
  faNode,
  faLaravel,
  faDocker,
  faFreebsd,
  faFedora,
  faJenkins,
  faWindows,
  faUbuntu,
  faSass,
  faStripe,
  faBlackberry,
  faGit,
  faDigitalOcean,
  faLinode,
} from "@fortawesome/free-brands-svg-icons";

// Initialize icon library to use throughout our app
library.add(
  faAngleDown,
  faAngleUp,
  faBadgePercent,
  faCommentAltLines,
  faEnvelope,
  faHeart,
  faMapMarkerAlt,
  faUser,
  faPaperPlane,
  faPhone,
  faFacebook,
  faFacebookF,
  faTwitter,
  faLinkedinIn,
  faReact,
  faNode,
  faLaravel,
  faDocker,
  faFreebsd,
  faFedora,
  faJenkins,
  faWindows,
  faUbuntu,
  faSass,
  faStripe,
  faBlackberry,
  faGit,
  faDigitalOcean,
  faLinode
);
