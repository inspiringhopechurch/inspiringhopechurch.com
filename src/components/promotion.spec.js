import React from "react";
import { mount, shallow } from "enzyme";
import Promotion from "./promotion";
// Import fontawesome library files
import "./icons";

describe("Promotion", () => {
  const discount = 5,
    date_ = "October 5, 2052 23:59:59",
    passedDate_ = "February 5, 2019 23:59:59";

  it("should render without crashing", () => {
    const promoBlock = shallow(
      <Promotion promoEndDate={date_} promoDiscount={discount} />
    );
    expect(promoBlock).toMatchSnapshot();
  });

  it("should render nothing without crashing", () => {
    const promoBlock = shallow(
      <Promotion promoEndDate={passedDate_} promoDiscount={discount} />
    );
    expect(promoBlock).toMatchSnapshot();
  });

  it("has the expected props when initially rendered", () => {
    const promoBlock = mount(
      <Promotion promoEndDate={date_} promoDiscount={discount} />
    );

    //expect(promoBlock.find(".notification")).toHaveLength(1);
    expect(promoBlock.prop("promoEndDate")).toBe(date_);
    expect(promoBlock.prop("promoEndDate")).not.toBe(passedDate_);
    expect(promoBlock.prop("promoDiscount")).toBe(discount);

    expect(promoBlock.find(".column.is-one-quarter h1")).toHaveLength(1);
    expect(promoBlock.find(".column.is-half")).toHaveLength(1);
  });
});
