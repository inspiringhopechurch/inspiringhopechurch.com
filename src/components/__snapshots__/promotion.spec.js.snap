// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Promotion should render nothing without crashing 1`] = `""`;

exports[`Promotion should render without crashing 1`] = `
<div
  className="notification is-link"
>
  <div
    className="columns is-vcentered is-centered"
  >
    <div
      className="column is-one-quarter"
    >
      <h1
        className="title has-text-centered is-1"
      >
        5
         
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Array [
              "fas",
              "badge-percent",
            ]
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
         Off!
      </h1>
    </div>
    <div
      className="column is-half"
    >
      Sample promo goes on until end of 
      October
      ! Register now for a 
      5
      %!
    </div>
  </div>
</div>
`;
